mlir/include/mlir/IR/PatternMatch.h

```diff
@@ -237,6 +237,9 @@ class Pattern {
 namespace detail {
 /// Helper class that derives from a RewritePattern class and provides separate
 /// `match` and `rewrite` entry points instead of a combined `matchAndRewrite`.
+///
+/// This class is deprecated. Use `matchAndRewrite` instead of separate `match`
+/// and `rewrite`.
```

Comment:
Do you plan to add a deprecated attribute around this in the future?

Reply:
I tried that, but it gives warnings because `matchAndRewrite` in this file calls `match` and `rewrite`.